src/managers/KeybindManager.hpp

```diff
@@ -95,6 +95,12 @@ class CKeybindManager {
 
     std::list<SKeybind>                                               m_lKeybinds;
 
+    //since we cant find keycode through keyname in xkb:
+    //on sendshortcut call, we once search for keyname (e.g. "g") the correct keycode (e.g. 42)
+    //and cache it in this map to make sendshortcut calls faster
+    //we also store the keyboard pointer (in the string) to differentiate between different keyboard (layouts)
+    std::unordered_map<std::string, xkb_keycode_t> m_mKeyToCodeCache;
```

Comment:
this cache needs to be cleared on a keyboard's keymap change. Easiest solution is to just clear it in .keymap of any kb

Reply:
better place would be the .keyboard listener in setupKeyboard in CInputManager

Reply:
Done, please check the location, I put it in the if block